### ISSUE TITLE
RichText page link rendering fails on non-default locales when not using WAGTAIL_I18N_ENABLED

### ISSUE THREAD

**Issue by sfriedel** [wagtail/wagtail#6540]
### Issue Summary

With WAGTAIL_I18N_ENABLED = False, rendering of RichText fields containing page links fails with a LookupError when a locale other than LANGUAGE_CODE is active.

### Steps to Reproduce

1. Start a new project with `wagtail start myproject`
2. `./manage.py migrate` / `./manage.py createsuperuser`
3. Add a LANGUAGES setting to myproject/settings/base.py:

    ```
    LANGUAGES = (
        ("en", "English"),
        ("de", "German"),
    )
    ```
4. Replace `home/models.py`:

    ```py
    from django.db import models
    from django.utils.translation import activate

    from wagtail.admin.edit_handlers import FieldPanel
    from wagtail.core.fields import RichTextField
    from wagtail.core.models import Page


    class HomePage(Page):
        text = RichTextField(blank=True)

        content_panels = Page.content_panels + [
            FieldPanel('text'),
        ]

        def serve(self, request):
            activate('de')
            return super().serve(request)
    ```
5. Replace `home/templates/home/welcome_page.html`

    ```
    {% load wagtailcore_tags %}

    {{ page.text|richtext }}
    ```
6. `./manage.py makemigrations` && `./manage.py migrate`
7. Run in `./manage.py shell`:

    ```py
    from home.models import HomePage
    p = HomePage.objects.first()
    p.text = f'<p><a id="{p.pk}" linktype="page">Home</a></p>'
    p.save_revision().publish()
    ```
8. Visit http://localhost:8000/ and receive the error:

    ```
    Traceback (most recent call last):
    File "/home/sef/.virtualenvs/wagtail-bugreport-u47kcAMn/lib/python3.8/site-packages/django/core/handlers/exception.py", line 47, in inner
        response = get_response(request)
    File "/home/sef/.virtualenvs/wagtail-bugreport-u47kcAMn/lib/python3.8/site-packages/django/core/handlers/base.py", line 179, in _get_response
        response = wrapped_callback(request, *callback_args, **callback_kwargs)
    File "/home/sef/.virtualenvs/wagtail-bugreport-u47kcAMn/lib/python3.8/site-packages/wagtail/core/views.py", line 17, in serve
        page, args, kwargs = site.root_page.localized.specific.route(request, path_components)
    File "/home/sef/.virtualenvs/wagtail-bugreport-u47kcAMn/lib/python3.8/site-packages/wagtail/core/models.py", line 1229, in localized
        localized = self.localized_draft
    File "/home/sef/.virtualenvs/wagtail-bugreport-u47kcAMn/lib/python3.8/site-packages/wagtail/core/models.py", line 1212, in localized_draft
        locale = Locale.get_active()
    File "/home/sef/.virtualenvs/wagtail-bugreport-u47kcAMn/lib/python3.8/site-packages/wagtail/core/models.py", line 372, in get_active
        return cls.objects.get_for_language(translation.get_language())
    File "/home/sef/.virtualenvs/wagtail-bugreport-u47kcAMn/lib/python3.8/site-packages/wagtail/core/models.py", line 339, in get_for_language
        return self.get(language_code=get_supported_content_language_variant(language_code))
    File "/home/sef/.virtualenvs/wagtail-bugreport-u47kcAMn/lib/python3.8/site-packages/wagtail/core/utils.py", line 271, in get_supported_content_language_variant
        raise LookupError(lang_code)
    LookupError: de
    ```

The error looks very similar to the one in #6510

* I have confirmed that this issue can be reproduced as described on a fresh Wagtail project: yes

### Technical details

* Python version: 3.8.5
* Django version: 3.1.3
* Wagtail version: 2.11.1
* Browser version: Chrome 86

**Comment by gasman** [comment]
Thanks for the detailed report @sfriedel! I've now opened a fix at #6549.

**Comment by sfriedel** [comment]
Awesome, thank you @gasman
